li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/FederatedSubscriptionState.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2019 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License").  See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.clients.consumer;
```

Comment:
kafka does not allow mixing multiple assignment types (i think?) yet this class does. would it be "safer" to protect this class from mixing multiple assignment types? even better - could this be made a parent interface with the distinct assignment types being concrete classes with final fields ?

Reply:
Done